code/modules/mob/living/basic/space_fauna/revenant/_revenant.dm

```diff
@@ -0,0 +1,513 @@
+	maximum_survivable_temperature = INFINITY
+
+	status_flags = NONE
+	density = FALSE
+	move_resist = MOVE_FORCE_OVERPOWERING
+	mob_size = MOB_SIZE_TINY
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	speed = 1
+	unique_name = TRUE
+	hud_possible = list(ANTAG_HUD)
+	hud_type = /datum/hud/revenant
+
+	/// The icon we use while just floating around.
+	var/icon_idle = "revenant_idle"
+	/// The icon we use while in a revealed state.
+	var/icon_reveal = "revenant_revealed"
+	/// The icon we use when stunned (temporarily frozen)
+	var/icon_stun = "revenant_stun"
+	/// The icon we use while draining someone.
+	var/icon_drain = "revenant_draining"
+
+	/// Are we currently dormant (ectoplasm'd)?
+	var/dormant = FALSE
+	/// If the revenant can take damage from normal sources.
+	var/revealed = FALSE
+	/// If the revenant's abilities are blocked by a holy power.
+	var/inhibited = FALSE
+	/// Are we currently draining someone?
+	var/draining = FALSE
+	/// Have we already given this revenant abilities?
+	var/generated_objectives_and_spells = FALSE
+
+	/// Lazylist of drained mobs to ensure that we don't steal a soul from someone twice
+	var/list/drained_mobs = null
+
+	/// The resource, and health, of revenants.
+	var/essence = 75
+	/// The regeneration cap of essence (go figure); regenerates every Life() tick up to this amount.
+	var/max_essence = 75
+	/// If the revenant regenerates essence or not
+	var/essence_regenerating = TRUE
+	/// How much essence regenerates per second
+	var/essence_regen_amount = 2.5
+	/// How much essence the revenant has stolen
+	var/essence_accumulated = 0
+	/// How much stolen essence is available for unlocks
+	var/essence_excess = 0
+	/// How long the revenant is revealed for, is about 2 seconds times this var.
+	var/unreveal_time = 0
+	/// How long the revenant is stunned for, is about 2 seconds times this var.
+	var/unstun_time = 0
+	/// How many perfect, regen-cap increasing souls the revenant has. //TODO, add objective for getting a perfect soul(s?)
+	var/perfectsouls = 0
+
+/mob/living/basic/revenant/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/simple_flying)
+	add_traits(list(TRAIT_SPACEWALK, TRAIT_SIXTHSENSE, TRAIT_FREE_HYPERSPACE_MOVEMENT), INNATE_TRAIT)
+
+	// Starting spells
+
+	var/datum/action/cooldown/spell/list_target/telepathy/revenant/telepathy = new(src)
+	telepathy.Grant(src)
+
+	// Starting spells that start locked
+	var/datum/action/cooldown/spell/aoe/revenant/overload/lights_go_zap = new(src)
+	lights_go_zap.Grant(src)
+
+	var/datum/action/cooldown/spell/aoe/revenant/defile/windows_go_smash = new(src)
+	windows_go_smash.Grant(src)
+
+	var/datum/action/cooldown/spell/aoe/revenant/blight/botany_go_mad = new(src)
+	botany_go_mad.Grant(src)
+
+	var/datum/action/cooldown/spell/aoe/revenant/malfunction/shuttle_go_emag = new(src)
+	shuttle_go_emag.Grant(src)
+
+	var/datum/action/cooldown/spell/aoe/revenant/haunt_object/toolbox_go_bonk = new(src)
+	toolbox_go_bonk.Grant(src)
```

Comment:
We'd save a handful of lines by doing this as a loop, maybe you could save some future dev adding a new ability (or revenant subtype?) some time by just letting them tack it on a list.

Reply:
noooo the sovl of the code i didn't even write